src/utils/validateIdentifierInputs.ts

```diff
@@ -0,0 +1,49 @@
+import { IdentifierInput } from "../types";
+import { fullPhoneNumberValidator } from "./validatePhoneNumbers";
+
+const isMatchRegex = (text: string, regex?: string): boolean => {
+  if (!regex) {
+    return true;
+  }
+  return new RegExp(regex).test(text);
+};
+
+const isUniqueList = (list: string[]): boolean =>
+  new Set(list).size === list.length;
+
+export const validateIdentifierInputs = (
+  identifierInputs: IdentifierInput[]
+): boolean => {
+  for (const { value, validationRegex, textInputType } of identifierInputs) {
+    if (!value) {
+      throw new Error(
+        `Please enter ${
+          identifierInputs.length === 1 ? "" : "unique "
+        }details to checkout`
+      );
+    }
+    if (textInputType === "NUMBER" && isNaN(Number(value))) {
+      throw new Error("Invalid details");
```

Comment:
small nit, i was wondering whether it would be clearer if it's `Invalid field value`, but its ok HAHA, merged liao 😂

Reply:
Oops sorryy, I followed the "details" copy that is currently in the app (when you don't fill in, it says "Please enter details to check out") But will check with Eida again

Reply:
HAHA no lah, dont need to say sorry haha. Okk~~ \^^/

Reply:
Eida will create separate card for all the different copy changes for prioritisation~